tests/rust_syscall_tests/src/lib.rs

```diff
@@ -0,0 +1,516 @@
+// SPDX-License-Identifier: Apache-2.0
+#![no_std]
+
+pub use sallyport::libc;
+
+use core::arch::asm;
+
+pub type Result<T> = core::result::Result<T, i32>;
+
+#[macro_export]
+macro_rules! startup {
+    () => {
+        #[no_mangle]
+        #[naked]
+        pub unsafe extern "sysv64" fn _start() -> ! {
+            unsafe fn start_c(dynv: *const u64, sp: *const usize) -> ! {
+                use goblin::elf64::program_header::{ProgramHeader, PT_DYNAMIC};
+                const AT_PHDR: usize = 3;
+                const AT_PHENT: usize = 4;
+                const AT_PHNUM: usize = 5;
+
+                let argc: usize = *sp;
+                let argv = sp.add(1);
+                let mut i = argc + 1;
+                while *argv.add(i) != 0 {
+                    i += 1;
+                }
+
+                let mut phnum: usize = 0;
+                let mut phentsize: usize = 0;
+                let mut ph: usize = 0;
+
+                let auxv_ptr = argv.add(i + 1);
+                let mut i = 0;
+                while *auxv_ptr.add(i) != 0 {
+                    match *auxv_ptr.add(i) {
+                        AT_PHDR => ph = *auxv_ptr.add(i + 1),
+                        AT_PHENT => phentsize = *auxv_ptr.add(i + 1),
+                        AT_PHNUM => phnum = *auxv_ptr.add(i + 1),
+                        _ => {},
+                    }
+                    i += 2;
+                }
+
+                let mut ph = ph as *const ProgramHeader;
+                let mut i = phnum;
+
+                while i != 0 {
```

Comment:
You can use `loop` here to avoid the `unreachable!()` after this.

Reply:
I want to stop looping if `i == 0` and panic if the if case was never reached.